spectrum-cardano-lib/src/types.rs

```diff
@@ -33,6 +33,22 @@ where
     }
 }
 
+impl<T> TryFromPData for Vec<T>
+where
+    T: TryFromPData,
+{
+    fn try_from_pd(data: PlutusData) -> Option<Self> {
+        data.into_vec().and_then(|pd_vec| {
+            pd_vec.iter().try_fold(vec![], |mut acc, raw_pd| {
+                T::try_from_pd(raw_pd.clone()).map(|parsed_value| {
```

Comment:
avoid cloning here, you can consume pd_vec at once, so no need to keep copies of elements

Reply:
Try pd_vec.into_iter()